controller/src/main/java/org/apache/rocketmq/controller/impl/DefaultBrokerHeartbeatManager.java

```diff
@@ -112,6 +115,24 @@ public void registerBroker(String clusterName, String brokerName, String brokerA
         }
     }
 
+    public boolean isBrokerActiveOfBrokerName(String clusterName, String brokerName) {
+
+        Set<BrokerAddrInfo> addrInfoSet = this.brokerLiveTable.keySet().stream()
+            .filter(item -> StringUtils.equals(clusterName, item.getClusterName())).collect(Collectors.toSet());
+        if (null == addrInfoSet || addrInfoSet.size() == 0) {
+            return false;
+        }
+        for (BrokerAddrInfo addrInfo : addrInfoSet) {
+            BrokerLiveInfo brokerLiveInfo = this.brokerLiveTable.get(addrInfo);
+            if (brokerLiveInfo != null && StringUtils.equals(brokerLiveInfo.brokerName, brokerName)) {
+                long last = brokerLiveInfo.lastUpdateTimestamp;
+                long timeoutMillis = brokerLiveInfo.heartbeatTimeoutMillis;
+                return (last + timeoutMillis) >= System.currentTimeMillis();
+            }
+        }
+        return false;
+    }
```

Comment:
Is it necessary to add this function? Maybe we can do this in replicasInfoManager

Reply:
> Is it necessary to add this function? Maybe we can do this in replicasInfoManager

I will do this in replicasInfoManager, and judge by the SyncStateInfo.masterAddress, I will get SyncStateInfo from syncStateSetInfoTable by brokerName. then if SyncStateInfo.masterAddress is not empty, at least one broker with the same BrokerName is alive